app/models/view_model/auctions_index.rb

```diff
@@ -0,0 +1,33 @@
+module ViewModel
+  class AuctionsIndex < Struct.new(:current_user, :auctions_query)
+    def auctions
+      @auctions ||= auctions_query.map  {|auction| Presenter::Auction.new(auction) }
+    end
+
+    def active_auction_count
+      auctions.count { |i| i.start_datetime < Time.now && Time.now < i.end_datetime }
+    end
+
+    def upcoming_auction_count
+      auctions.count { |i| Time.now < i.start_datetime }
+    end
+
+    def header_partial
+      if current_user && current_user.sam_account?
+        '/components/sam_verified_header'
```

Comment:
I think these are view models, which is to say they are tightly bound to a single view. In this case it is /auctions or the AuctionsController#index action.

Components seems like a good place for the view things that span concepts. Sometimes I also see these partials hanging out inside the layouts directory of views.